### PR TITLE
Refactored /web shared files (middleware and utils) to new shared folder

### PR DESCRIPTION
**Pull request by rshbhgrg**
refs #9866 

- Moves web/middleware -> web/shared/middleware
- Moves web/utils.js -> web/shared/utils.js

### CODE REVIEW

File: core/server/api/authentication.js [hunk 1]
```diff
@@ -327,7 +327,7 @@ authentication = {
                     return updatedUser.save(options);
                 })
                 .catch(common.errors.ValidationError, (err) => {
-                    return Promise.reject(err); 
+                    return Promise.reject(err);
```
Comment (kirrg001) [inline core/server/api/authentication.js]: Why is this happening? 🤔

File: core/server/api/authentication.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const Promise = require('bluebird'),
     urlService = require('../services/url'),
     localUtils = require('./utils'),
     models = require('../models'),
-    spamPrevention = require('../web/middleware/api/spam-prevention'),
+    spamPrevention = require('../web/shared/middleware/api/spam-prevention'),
```
Comment (kirrg001) [inline core/server/api/authentication.js]: Did you consider renaming this to `../shared/middlewares/...`? What are your thoughts?

Comment (rishabhgrg) [reply core/server/api/authentication.js]: Yeah +1, I think should be `middlewares` as well, updating

Comment (kirrg001) [reply core/server/api/authentication.js]: Cool :)

File: core/server/api/authentication.js (hunk 1 above)
Comment (rishabhgrg) [reply core/server/api/authentication.js]: We had an extra space after semi-colon in this code, came with merge of `eslint-plugin-ghost` changes it seems, apparently lint didn't catch and I inadvertently fixed it while updating this file 🙂

Comment (kirrg001) [reply core/server/api/authentication.js]: Ok nice.

Looks like https://eslint.org/docs/rules/no-trailing-spaces is not defined in our plugin 🤔 Worth raising an issue. You wanne do it?

Comment (rishabhgrg) [reply core/server/api/authentication.js]: Yeah on it 🏃

**Review by kirrg001 — APPROVED** [review]
👍

**Pull request merged**